src/test/java/calculator/calculator/CalculatorTest.java

```diff
@@ -0,0 +1,106 @@
+package calculator.calculator;
+
+import calculator.calculator.Calculator;
+import calculator.exception.ErrorMessage;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.*;
+
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.*;
+
+public class CalculatorTest {
+
+    private Calculator calculator;
+
+    @BeforeEach
+    void setUp() {
+        calculator = Calculator.of();
+    }
+
+    @DisplayName("Calculator 생성 확인")
+    @Test
+    void create() {
+        assertThatCode(Calculator::of).doesNotThrowAnyException();
+    }
+
+    @DisplayName("입력 값이 null 이거나 빈 공백 문자일 경우 IllegalArgumentException 에러 발생")
+    @ParameterizedTest
+    @NullAndEmptySource
+    void isNullOrEmpty(final String value) {
+        assertThatExceptionOfType(IllegalArgumentException.class)
+                .isThrownBy(() -> calculator.run(value))
+                .withMessageContaining(ErrorMessage.IS_NULL_OR_EMPTY);
+    }
+
+    @DisplayName("0을 입력하면 0을 반환")
+    @Test
+    void zeroValue() {
+        assertThat(calculator.run("0")).isZero();
+    }
+
+    @DisplayName("덧셈 테스트")
+    @ParameterizedTest
+    @CsvSource(value = {"2 + 3=5", "1 + 1 + 10000=10002", "0 + 999=999", "1=1"}, delimiter = '=')
+    void plus(final String value, final int expected) {
+        assertThat(calculator.run(value)).isEqualTo(expected);
+    }
+
+    @DisplayName("뺄셈 테스트")
+    @ParameterizedTest
+    @CsvSource(value = {"3 - 2=1", "10000 - 9999=1", "-100 - -50=-50", "3 - 100=-97"}, delimiter = '=')
+    void minus(final String value, final int expected) {
+        assertThat(calculator.run(value)).isEqualTo(expected);
+    }
+
+    @DisplayName("곱셈 테스트")
+    @ParameterizedTest
+    @CsvSource(value = {"3 * 2=6", "-1 * 2=-2", "-100 * -50=5000"}, delimiter = '=')
+    void times(final String value, final int expected) {
+        assertThat(calculator.run(value)).isEqualTo(expected);
+    }
+
+    @DisplayName("나눗셈 테스트")
+    @ParameterizedTest
+    @CsvSource(value = {"4 / 2=2", "40000 / 1=40000", "0 / 20=0"}, delimiter = '=')
+    void divide(final String value, final int expected) {
+        assertThat(calculator.run(value)).isEqualTo(expected);
+    }
+
+    @DisplayName("기본적인 복합 연산 확인")
+    @ParameterizedTest
+    @MethodSource("basicCase")
+    void success(final String value, final int expected) {
+        assertThat(calculator.run(value)).isEqualTo(expected);
+    }
+
+    private static Stream<Arguments> basicCase() {
```

Comment:
이미 알고 계실수도 있겠지만
@MethodSource의 값으로 함수명을 기입하지 않고, 동일한 함수명을 사용할 수 있어요
```suggestion
    @ParameterizedTest
    @MethodSource
    void success(final String value, final int expected) {
        assertThat(calculator.run(value)).isEqualTo(expected);
    }

    private static Stream<Arguments> success() {
        ...
    }
```

Reply:
넵!! 알고 있었지만 remind 감사합니다 ㅎㅎ